sentry/src/test/java/io/sentry/TracesSamplerTest.kt

```diff
@@ -136,23 +243,55 @@ class TracesSamplerTest {
         )
         assertTrue(samplingDecisionParentSampled.sampled)
         assertNull(samplingDecisionParentSampled.sampleRate)
+        assertTrue(samplingDecisionParentSampled.profileSampled)
+        assertNull(samplingDecisionParentSampled.profileSampleRate)
+    }
+
+    @Test
+    fun `when parentSampled is not set and parentProfileSampled is set, profile is not sampled`() {
+        val sampler = fixture.getSut()
+        val transactionContextParentUnsampled = TransactionContext("name", "op")
+        transactionContextParentUnsampled.setParentSampled(false, true)
+        val samplingDecisionParentSampled = sampler.sample(
+            SamplingContext(
+                transactionContextParentUnsampled,
+                CustomSamplingContext()
+            )
+        )
+        assertFalse(samplingDecisionParentSampled.sampled)
+        assertNull(samplingDecisionParentSampled.sampleRate)
+        assertFalse(samplingDecisionParentSampled.profileSampled)
+        assertNull(samplingDecisionParentSampled.profileSampleRate)
     }
 
     @Test
     fun `when tracing decision is set on SpanContext, sampler uses it as a sampling decision`() {
         val sampler = fixture.getSut()
         val transactionContextNotSampled = TransactionContext("name", "op")
-        transactionContextNotSampled.sampled = false
+        transactionContextNotSampled.setSampled(false, false)
         val samplingDecision =
             sampler.sample(SamplingContext(transactionContextNotSampled, CustomSamplingContext()))
         assertFalse(samplingDecision.sampled)
         assertNull(samplingDecision.sampleRate)
+        assertFalse(samplingDecision.profileSampled)
+        assertNull(samplingDecision.profileSampleRate)
 
         val transactionContextSampled = TransactionContext("name", "op")
-        transactionContextSampled.sampled = true
+        transactionContextSampled.setSampled(true, true)
```

Comment:
Isn't that a breaking change in the public API?

Reply:
not really, as profile sampled defaults to false so 
```
        transactionContextNotSampled.sampled = false
        transactionContextNotSampled.setSampled(false, false) 
```
are the same, like
```
        transactionContextSampled.sampled = true
        transactionContextSampled.setSampled(true, false)
```
the `setSampled(..., true)` is the real change, but it's an addition, rather than a change. 
I can change `setSampled(false, false)` to `sampled = false` in the test for clarity

Reply:
If `setSampled` is a new API, its ok.
I thought that we replaced a public API to another.